Support for Conditional Compilation

vannuysm opened the pull request:
Add support for conditional compilation as per #60

nishtahir commented:
A working sample needs to go into the `sample-project`

loopyDeveloper commented on grammar/BrightScript.g4:
```diff
@@ -50,6 +51,15 @@ associativeElementInitializer
+    | CONDITIONAL_ERROR .*?
+    | CONDITIONAL_IF expression
+    | CONDITIONAL_ELSEIF expression
+    | CONDITIONAL_ELSE expression
```
@vannuysm 

maybe else should not have an expression?

vannuysm replied:
Good catch!

loopyDeveloper commented on grammar/BrightScript.g4:
```diff
@@ -558,6 +568,32 @@ WS
+    : '#' ELSE
+    ;
+
+CONDTIONAL_ELSEIF
```
typo spelling here

should be CONDITIONAL_ELSEIF

vannuysm replied:
Some day I'll make my 8th grade typing teacher proud. Today is not that day. 😃

vannuysm commented:
I have added a couple of test scenarios to the sample project and fixed the typos and such. I feel comfortable enough to remove the `DO NOT MERGE` label from this now.
